src/main/java/com/eerussianguy/blazemap/feature/waypoints/WaypointRenderer.java

```diff
@@ -120,29 +124,38 @@ private static void renderWaypoint(Minecraft mc, PoseStack stack, MultiBufferSou
      */
     private static void renderLabel(Minecraft mc, PoseStack stack, MultiBufferSource.BufferSource buffers, Waypoint w, Vec3 pos) {
         stack.pushPose();
+
         float width = 32;
         float height = 32;
+
         translateFromCameraToPos(stack, pos);
-        stack.mulPoseMatrix(new Matrix4f(mc.gameRenderer.getMainCamera().rotation()));
+        stack.mulPose(mc.gameRenderer.getMainCamera().rotation());
```

Comment:
I don't think this is _quite_ working right, but given in-world waypoints are turned off at the moment due to existing rendering issues, I can sort out the details when I resume BME-46